src/SignhostAPIClient/Rest/SignHostApiClient.cs

```diff
@@ -110,17 +132,46 @@ public async Task<ApiResponse<Transaction>> GetTransactionResponseAsync(string t
 
 		/// <inheritdoc />
 		public async Task<Transaction> GetTransactionAsync(string transactionId)
+			=> await GetTransactionAsync(transactionId, default)
+				.ConfigureAwait(false);
+
+		/// <inheritdoc />
+		public async Task<Transaction> GetTransactionAsync(
+			string transactionId,
+			CancellationToken cancellationToken = default)
 		{
-			var response = await GetTransactionResponseAsync(transactionId)
+			var response = await GetTransactionResponseAsync(
+				transactionId,
+				cancellationToken)
 				.ConfigureAwait(false);
 
 			response.EnsureAvailableStatusCode();
 
 			return response.Value;
 		}
 
+		public async Task DeleteTransactionAsync(
```

Comment:
Why do we need this method 3 times? Also, shouldn't DeleteTransactionOptions stay optional.

I think we need 1. `DeleteTransactionAsync(string, DeleteTransactionOptions = null)` and 2. `DeleteTransactionAsync(string, DeleteTransactionOptions = null, CancellationToken = default)`

Reply:
The problem here is that when you make a call as follows DeleteTransactionAsync("id") then the compiler doesn't know which method to choose.

Reply:
Yes, I suppose so.